scripts/emerge/stage_prs_anvil_deliverable_data.py

```diff
@@ -0,0 +1,193 @@
+                                       "crsp_sample_id"]]
+    # rename entity column and make entity:entity_id column first in tsv
+    anvil_arrays_df = anvil_arrays_df.rename(columns={"crsp_sample_id": "entity:Arrays_id"})
+    first_column = anvil_arrays_df.pop("entity:Arrays_id")
+    anvil_arrays_df.insert(0, "entity:Arrays_id", first_column)
+
+    # Polygenic Risk Scores
+    # create prs_scores subset df
+    anvil_prs_scores_df = prs_entities_df[["ast_raw", "ast_adjusted", "ast_percentile", "brca_raw", "brca_adjusted",
+                                           "brca_percentile", "afib_raw", "afib_adjusted", "afib_percentile",
+                                           "chd_raw", "chd_adjusted", "chd_percentile", "ckd_raw", "ckd_adjusted",
+                                           "ckd_percentile", "hcl_raw", "hcl_adjusted", "hcl_percentile", "bmi_raw",
+                                           "bmi_adjusted", "bmi_percentile", "prca_raw", "prca_adjusted", "prca_percentile",
+                                           "t1d_raw", "t1d_adjusted", "t1d_percentile", "t2d_raw", "t2d_adjusted", "t2d_percentile",
+                                           "crsp_sample_id"]]
+    # rename entity column and make entity:entity_id column first in tsv
+    anvil_prs_scores_df = anvil_prs_scores_df.rename(columns={"crsp_sample_id": "entity:Polygenic_Risk_Scores_id"})
+    first_column = anvil_prs_scores_df.pop("entity:Polygenic_Risk_Scores_id")
+    anvil_prs_scores_df.insert(0, "entity:Polygenic_Risk_Scores_id", first_column)
+
+    # write individual dataframes to tsv files
+    anvil_arrays_df.to_csv("arrays.tsv", sep = "\t", index=False)
+    anvil_samples_df.to_csv("samples.tsv", sep="\t", index=False)
+    anvil_prs_scores_df.to_csv("prs_scores.tsv", sep="\t", index=False)
+
+    # load tsv files to destination workspace
+    for tsv in ["arrays.tsv", "samples.tsv", "prs_scores.tsv"]:
+        print(f"Loading {tsv} to {dest_workspace}.")
+        upload_data_table(tsv, dest_workspace, dest_namespace)
+
+
+def copy_object_to_bucket(src_bucket_name, src_object_name, dest_bucket_name, dest_object_name):
+    """Copies object from one bucket to another with a new name."""
+
+    storage_client = gcs.Client()
+
+    source_bucket = storage_client.bucket(src_bucket_name)
+    source_object = source_bucket.blob(src_object_name)
+    destination_bucket = storage_client.bucket(dest_bucket_name)
+
+    blob_copy = source_bucket.copy_blob(source_object, destination_bucket, dest_object_name)
+
+
+def rename_and_rehome_data_files(prs_entities_dicts, dest_bucket, snapshot_id):
+    """For columns in the df that represent files, update paths pointing to the final bucket and copy files."""
+
+    # file types and their file extensions to use for renaming files in delivery workspace
+    file_cols = {"red_idat_cloud_path":"_Red.idat", "green_idat_cloud_path":"_Grn.idat",
```

Comment:
minor: should this variable be named `file_extensions` ? `_cols` suggests "columns" to me